elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAutoConfiguration.java

```diff
@@ -90,32 +127,66 @@ public void inject(Object entity) {
                 });
 
         dictionary.scanForSecurityChecks();
+
+        if (settings.getDynamicConfig().isEnabled()) {
+            ElideDynamicEntityCompiler compiler = dynamicCompiler.getIfAvailable();
+            Set<Class<?>> annotatedClass = findAnnotatedClasses(compiler, SecurityCheck.class);
+            dictionary.addSecurityChecks(annotatedClass);
+        }
+
         return dictionary;
     }
 
     /**
      * Create a QueryEngine instance for aggregation data store to use.
      * @param entityManagerFactory The JPA factory which creates entity managers.
+     * @param dynamicCompiler An instance of objectprovider for ElideDynamicEntityCompiler.
+     * @param settings Elide configuration settings.
      * @return An instance of a QueryEngine
+     * @throws ClassNotFoundException
      */
     @Bean
     @ConditionalOnMissingBean
-    public QueryEngine buildQueryEngine(EntityManagerFactory entityManagerFactory) {
+    public QueryEngine buildQueryEngine(EntityManagerFactory entityManagerFactory,
+            ObjectProvider<ElideDynamicEntityCompiler> dynamicCompiler, ElideConfigProperties settings)
+            throws ClassNotFoundException {
+
         MetaDataStore metaDataStore = new MetaDataStore();
 
+        if (settings.getDynamicConfig().isEnabled()) {
+            ElideDynamicEntityCompiler compiler = dynamicCompiler.getIfAvailable();
+            Set<Class<?>> annotatedClass = findAnnotatedClasses(compiler, FromTable.class);
+            annotatedClass.addAll(findAnnotatedClasses(compiler, FromSubquery.class));
+            metaDataStore.populateEntityDictionary(annotatedClass);
```

Comment:
Pass the compiler to the MetaDataStore constructor and initialize the internal dictionary in the constructor instead of doing this initialization here.